gnomad_qc/v3/create_release/prepare_vcf_data_release.py

```diff
@@ -0,0 +1,1019 @@
+    ht = ht.transmute(info=hl.struct(**info_annot_mapping))
+
+    logger.info("Adjusting VCF incompatiable types...")
+    # Reformat AS_SB_TABLE for use in ht_to_vcf_mt
+    ht = ht.annotate(
+        info=ht.info.annotate(
+            AS_SB_TABLE=hl.array([ht.info.AS_SB_TABLE[:2], ht.info.AS_SB_TABLE[2:]])
+        )
+    )
+    ht = adjust_vcf_incompatible_types(ht)
+
+    return ht, new_row_annots
+
+
+def build_parameter_dict(
+    hgdp_1kg_subset: bool = False, test: bool = False
+) -> Dict[str, Union[bool, str, List, Dict, Set, hl.Table, None]]:
+    """
+    Build a dictionary of the parameters needed in the script because they differ for the HGDP + TGP subset release
+    compared to the full release.
+
+    :param hgdp_1kg_subset: Build the parameter list specific to the HGDP + TGP subset release.
+    :param test: Uses a checkpoint path for the prepared VCF Table that is specific to testing.
+    :return: Dictionary containing parameters needed for making the release VCF.
+    """
+    if hgdp_1kg_subset:
+        parameter_dict = {
+            "pops": HGDP_KG_POPS,
+            "subsets": ["", "gnomad"],
+            "is_subset": True,
+            "temp_ht_path": get_checkpoint_path(
+                f"vcf_prep_{'test' if test else ''}_hgdp_1kg"
+            ),
+            "include_age_hists": False,
+            "subset_pops": {"gnomad": POPS, "": HGDP_KG_POPS},
+            "vcf_info_reorder": HGDP_KG_VCF_INFO_REORDER,
+            "ht": release_subset(subset="hgdp_1kg", dense=True).mt().rows(),
+            "freq_entries_to_remove": set(),
+            "age_hist_data": None,
+        }
+
+    else:
+        parameter_dict = {
+            "pops": POPS,
+            "subsets": SUBSET_LIST_FOR_VCF,
+            "is_subset": False,
+            "temp_ht_path": get_checkpoint_path(
+                f"vcf_prep_{'test' if args.test else ''}"
+            ),
+            "include_age_hists": True,
+            "subset_pops": {"hgdp": HGDP_POPS, "tgp": KG_POPS},
+            "vcf_info_reorder": VCF_INFO_REORDER,
+            "ht": hl.read_table(
+                release_ht_path()
+            ),  # TODO: Change to release_sites().ht(),
+        }
+        # Downsampling and subset entries to remove from VCF's freq export
+        # Note: Need to extract the non-standard downsamplings from the freq_meta struct to the FREQ_ENTRIES_TO_REMOVE
+        freq_entries_to_remove = {
+            str(x["downsampling"])
+            for x in hl.eval(parameter_dict["ht"].freq_meta)
+            if "downsampling" in x
+        }
+        freq_entries_to_remove.update(set(COHORTS_WITH_POP_STORED_AS_SUBPOP))
+        parameter_dict["freq_entries_to_remove"] = freq_entries_to_remove
+        parameter_dict["age_hist_data"] = hl.eval(parameter_dict["ht"].age_distribution)
+
+    return parameter_dict
+
+
+def main(args):
+
+    hl.init(
+        log="/vcf_release.log",
+        default_reference="GRCh38",
+        tmp_dir="hdfs:///vcf_write.tmp/",
```

Comment:
This may be updated by now, I've just always added it out of habit. Historically you needed it to use the SSDs but that may not be the case anymore: https://atgu.slack.com/archives/CNE92QZMF/p1542132816013200